File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<int> prime(n+1,1);
        for(int i = 2;(i*i)<=n;i++)
        {
            if(prime[i]==1)
            {
                for(int k = i*i;k<=n;k+=i)
                 prime[k]=0;
            }
        }
        int cnt = 0;
        for(int i = 2;i<n;i++)
        {
            if(prime[i])
                cnt++;
        }
        return cnt;
    }
};

auto init = []
{
    ios::sync_with_stdio(NULL);
    cin.tie(NULL);
    cout.tie(NULL);
    return 'c';
}();

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        #Create a boolean list to track prime status of numbers
        prime = [True] * (n + 1)
        p = 2

        # Sieve of Eratosthenes algorithm
        while p * p <= n:
            if prime[p]:
                
                # Mark all multiples of p as non-prime
                for i in range(p * p, n + 1, p):
                    prime[i] = False
            p += 1

        # Collect all prime numbers
        count = 0
        for p in range(2, n):
            if prime[p]:
                count+=1
        
        return count

        